drivers/docker/driver.go

```diff
@@ -760,6 +760,12 @@ func (d *Driver) createContainerConfig(task *drivers.TaskConfig, driverConfig *T
 		}
 		hostConfig.Runtime = d.config.GPURuntimeName
 	}
+	if driverConfig.OCIRuntime != "" {
```

Comment:
Might worth checking if a user was explicit and set the gpu runtime here again.

```suggestion
	if driverConfig.OCIRuntime != "" && driverConfig.OCIRuntime != hostConfig.Runtime {
```